web_external/ClipExplorer/index.jsx

```diff
@@ -0,0 +1,227 @@
+import _ from 'underscore';
+import React, { Component } from 'react';
+import { Modal, ButtonToolbar, Button, Row, Col, Glyphicon, Checkbox } from 'react-bootstrap';
+import { restRequest } from 'girder/rest';
+
+import style from './style.styl';
+
+class ClipExplorer extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            paths: [{ name: 'root', folderId: 'root' }],
+            currentFolders: [],
+            rootFolders: null,
+            annotationFolders: new Set(),
+            selectedItem: null,
+            selectedFolderId: null,
+            showMode: 'all'
+        };
+        this.folderCache = new Map();
+    }
+
+    componentDidMount() {
+        this.fetchFolder();
+        this.getAllAnnotationFolders();
+    }
+
+    getAllFoldersAtRoot() {
+        return Promise.all([
+            restRequest({
+                url: '/collection',
+            }).then((collections) => {
+                return Promise.all(collections.map((collection) => {
+                    return restRequest({
+                        url: 'folder',
+                        data: {
+                            parentType: 'collection',
+                            parentId: collection._id
+                        }
+                    })
+                }));
+            })
+        ]).then((results) => {
+            var folders = _.flatten(results);
+            return this.attachImageItem(folders);
+        })
+    }
+
+    getAllAnnotationFolders() {
+        // The current set of API is lacking flexible search, this is the method to get all annotated video clips (folder)
```

Comment:
can we wrap this very long comment? I would suggest we stick to something reasonable (79, 120 or any other number). I am okay if we do in some other PR as well and can be convinced otherwise.

Reply:
I agree, but am actually somewhat inclined to punt on this; we need to agree on a style and trying to fix it now may duplicate work.

Reply:
Yes, I am looking into adding jsx eslinting to the project. and we will find out the standard for jsx out there.